From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KurumiFake <kurumiisshidowife@gmail.com>
Date: Sat, 19 Mar 2022 03:38:06 +0000
Subject: [PATCH] (Lazinity) Tighten NBT read limit

dual side dagger: whilst blocking big packet crashes, makes it easier to chunkban i guess
tighten your books.

// make the limit a lil bit higher here idk

diff --git a/src/main/java/net/minecraft/server/PacketDataSerializer.java b/src/main/java/net/minecraft/server/PacketDataSerializer.java
index c1273e988e6a46c20eca54530f928404c03ad947..d2f16c75b57cc82d7cb17051559450d282258b0a 100644
--- a/src/main/java/net/minecraft/server/PacketDataSerializer.java
+++ b/src/main/java/net/minecraft/server/PacketDataSerializer.java
@@ -228,7 +228,7 @@ public class PacketDataSerializer extends ByteBuf {
             this.readerIndex(i);
 
             try {
-                return NBTCompressedStreamTools.a((DataInput) (new ByteBufInputStream(this)), new NBTReadLimiter(2097152L));
+                return NBTCompressedStreamTools.a((DataInput) (new ByteBufInputStream(this)), new NBTReadLimiter(63000L));
             } catch (IOException ioexception) {
                 throw new EncoderException(ioexception);
             }
